GitCommands/Config/ConfigFile.cs

```diff
@@ -507,15 +507,9 @@ private ParsePart ReadValue(char c)
                             _escapedValue = true;
                             return ReadValue;
                         case '"':
-                            if (_quotedValue)
-                            {
-                                _token.Append(_valueToken);
-                            }
-                            else
-                            {
-                                _token.Append(_valueToken.ToString().Trim());
-                            }
-
+                            _token.Append(_quotedValue
+                                ? _valueToken.ToString()
+                                : _valueToken.ToString().Trim());
```

Comment:
This is some strange piece of code

Reply:
It does look a bit strange. These kinds of state-based lexers are very powerful but can be hard to reason about. Gotta get in the zone. And have lots of tests.